### PR TITLE
Adding Disperse

### PR DESCRIPTION
**Pull request by cphyc**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by cphyc** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/disperse/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+set -euo pipefail
+set -x
+
+mkdir -p build
+
+cd build
+
+cmake -DCMAKE_INSTALL_PREFIX=${PREFIX} ../
+make -j${CPU_COUNT}
+make install
```
Comment (cphyc) [inline recipes/disperse/build.sh:10]: Note: there is unfortunately no test bundled with this software.

File: recipes/disperse/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - gmp
+    - openmp  # [osx]
+    - _openmp_mutex  # [linux]
+
+test:
+  commands:
+    - mse
+    - fieldconv
+    - netconv
+    - skelconv
```
Comment (cphyc) [inline recipes/disperse/meta.yaml:35]: This is an exhaustive list of all the binaries generated by the compilation step.

**Review by cphyc — COMMENTED** [review]
Leaving reviews for @conda-forge/staged-recipes. The software may be installable on Windows, but I don't think this has been tested so I left it out for this PR.

File: recipes/disperse/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  summary: 'Automatic identification of persistent structures.'
+  description: |
+    DisPerSE stands for "Discrete Persistent Structures Extractor" and its main
+    purpose is to identify persistent topological features such as peaks, voids,
+    walls and in particular filamentary structures within sampled distributions
+    in 2D, 3D, and possibly more...
+  license: CECILL-C
+  license_family: GPL2
+  license_file: License.txt
+  doc_url: http://www2.iap.fr/users/sousbie/web/html/indexd41d.html?
```
Comment (ocefpaf) [inline recipes/disperse/meta.yaml:48]:
```suggestion
  doc_url: https://www2.iap.fr/users/sousbie/web/html/indexd41d.html
```

**Pull request merged**